src/document.c

```diff
@@ -757,10 +761,21 @@ static PreprocessorFunc preprocessorMap[] = {
     [IXFLDPOS_GEOMETRY] = geometryPreprocessor,
     };
 
+
 int IndexerBulkAdd(RSAddDocumentCtx *cur, RedisSearchCtx *sctx,
                    const DocumentField *field, const FieldSpec *fs, FieldIndexerData *fdata,
-                   QueryError *status) {
+                   QueryError *status) {  
+  
+  static size_t opCounter = 0;
+  // Yield to Redis every RSGlobalConfig.indexerYieldEveryOps operations
+  if (RedisModule_Yield && ++opCounter >= RSGlobalConfig.indexerYieldEveryOps) {
+    opCounter = 0;
+    IncrementYieldCounter(); // Track that we called yield
+    RedisModule_Yield(sctx->redisCtx, REDISMODULE_YIELD_FLAG_CLIENTS, NULL);
+  }
+  
```

Comment:
We want to have the opCounter before every indexing operation **per field** (not per document)